ProductComment.php

```diff
@@ -1,443 +1,459 @@
-		return $success;
-	}
-
-	/**
-	 * Delete a comment, grade and report data
-	 *
-	 * @return boolean succeed
-	 */
-	public function delete()
-	{
-		parent::delete();
-		ProductComment::deleteGrades($this->id);
-		ProductComment::deleteReports($this->id);
-		ProductComment::deleteUsefulness($this->id);
-	}
-
-	/**
-	 * Delete Grades
-	 *
-	 * @return boolean succeed
-	 */
-	public static function deleteGrades($id_product_comment)
-	{
-		if (!Validate::isUnsignedId($id_product_comment))
-			return false;
-		return (Db::getInstance()->execute('
-		DELETE FROM `'._DB_PREFIX_.'product_comment_grade`
-		WHERE `id_product_comment` = '.(int)$id_product_comment));
-	}
-
-	/**
-	 * Delete Reports
-	 *
-	 * @return boolean succeed
-	 */
-	public static function deleteReports($id_product_comment)
-	{
-		if (!Validate::isUnsignedId($id_product_comment))
-			return false;
-		return (Db::getInstance()->execute('
-		DELETE FROM `'._DB_PREFIX_.'product_comment_report`
-		WHERE `id_product_comment` = '.(int)$id_product_comment));
-	}
-
-	/**
-	 * Delete usefulness
-	 *
-	 * @return boolean succeed
-	 */
-	public static function deleteUsefulness($id_product_comment)
-	{
-		if (!Validate::isUnsignedId($id_product_comment))
-			return false;
-
-		return (Db::getInstance()->execute('
-		DELETE FROM `'._DB_PREFIX_.'product_comment_usefulness`
-		WHERE `id_product_comment` = '.(int)$id_product_comment));
-	}
-
-	/**
-	 * Report comment
-	 *
-	 * @return boolean
-	 */
-	public static function reportComment($id_product_comment, $id_customer)
-	{
-		return (Db::getInstance()->execute('
-			INSERT INTO `'._DB_PREFIX_.'product_comment_report` (`id_product_comment`, `id_customer`)
-			VALUES ('.(int)$id_product_comment.', '.(int)$id_customer.')'));
-	}
-
-	/**
-	 * Comment already report
-	 *
-	 * @return boolean
-	 */
-	public static function isAlreadyReport($id_product_comment, $id_customer)
-	{
-		return (bool)Db::getInstance()->getValue('
+		FROM `' . _DB_PREFIX_ . 'product_comment` pc
+		LEFT JOIN `' . _DB_PREFIX_ . 'customer` c ON (c.`id_customer` = pc.`id_customer`)
+		LEFT JOIN `' . _DB_PREFIX_ . 'product_lang` pl ON (pl.`id_product` = pc.`id_product` AND pl.`id_lang` = ' . (int) Context::getContext()->language->id . Shop::addSqlRestrictionOnLang('pl') . ')
+		ORDER BY pc.`date_add` DESC');
+    }
+
+    /**
+     * Validate a comment
+     *
+     * @return bool succeed
+     */
+    public function validate($validate = '1')
+    {
+        if (!Validate::isUnsignedId($this->id)) {
+            return false;
+        }
+
+        $success = (Db::getInstance()->execute('
+		UPDATE `' . _DB_PREFIX_ . 'product_comment` SET
+		`validate` = ' . (int) $validate . '
+		WHERE `id_product_comment` = ' . (int) $this->id));
+
+        Hook::exec('actionObjectProductCommentValidateAfter', array('object' => $this));
+
+        return $success;
+    }
+
+    /**
+     * Delete a comment, grade and report data
+     *
+     * @return bool succeed
```

Comment:
Missing return statement

Reply:
Same thing here, even though this one is still used by the back office.
I don't want to waste time improving this (soon to be removed) code (it works for now it is enough ^^)